### PR TITLE
Update arcade to Preview 5

### PR DESCRIPTION
**Pull request by mmitche**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/tools/dotnet-cli/DotNetCli.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <IncludeDotNetCli Condition=" '$(IncludeDotNetCli)' != 'true' ">false</IncludeDotNetCli>
     <!-- TODO (https://github.com/dotnet/arcade/issues/7787): When bumping this, we should do #7787 -->
-    <AspNetCoreRuntimeVersion>7.0.0-preview.2.22153.2</AspNetCoreRuntimeVersion>
+    <AspNetCoreRuntimeVersion>7.0.0-preview.4.22251.1</AspNetCoreRuntimeVersion>
```
Comment (MattGal) [inline src/Microsoft.DotNet.Helix/Sdk/tools/dotnet-cli/DotNetCli.props:5]: The above commented issue is long closed... :)

**Review by MattGal — DISMISSED** [review]

**Comment by ViktorHofer** [comment]
@mmitche Preview4 hasn't yet publicly released, has it? IIRC we agreed on in Tactics that we only depend on publicly released, signed and documented preview versions.

**Comment by adiaaida** [comment]
Looks like there is something in preview 4 that breaks our remote executor tests on redhat. We should probably look into what's going on there (logs suggest an AV). It'd be nice to get this revved since preview 2 is broken in some instances.

**Comment by mmitche** [comment]
@adiaaida Who is the right person to look at the remote executor failures?

**Comment by adiaaida** [comment]
It looks like @elinor-fung last touched those tests? I think RemoteExecutor is owned by @dotnet/runtime-infrastructure

**Comment by adiaaida** [comment]
I guess the stack traces are confusing then?

```
Starting:    Microsoft.DotNet.RemoteExecutor.Tests
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].ExecutionContextCallback(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].MoveNext(System.Threading.Thread)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].MoveNext()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Runtime.CompilerServices.IAsyncStateMachineBox, Boolean)
   at System.Threading.Tasks.Task.RunContinuations(System.Object)
   at System.Threading.Tasks.Task.TrySetResult()
   at System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()
   at System.Threading.TimerQueueTimer.Fire(Boolean)
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.UnmanagedThreadPoolWorkItem.System.Threading.IThreadPoolWorkItem.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
   at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncFunc_ThrowException>b__3_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 54
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 61
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
**Expected: True
Actual:   False**
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
   at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncAction_ThrowException>b__1_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 29
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 65
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c.<FatalError_AV>b__8_1()
   at System.RuntimeMethodHandle.InvokeMethod(System.Object, Void**, System.Signature, Boolean)
   at System.Reflection.MethodInvoker.Invoke(System.Object, IntPtr*, System.Reflection.BindingFlags)
   at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
   at Microsoft.DotNet.RemoteExecutor.Program.Main(System.String[])
  Finished:    Microsoft.DotNet.RemoteExecutor.Tests
```

**Comment by mmitche** [comment]
> Hm. The stack trace is definitely troubling to me and it seems like it's saying that it expects true but gets false.
> 
> ```
> Starting:    Microsoft.DotNet.RemoteExecutor.Tests
> Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
>    at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d.MoveNext()
>    at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].ExecutionContextCallback(System.Object)
>    at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
>    at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].MoveNext(System.Threading.Thread)
>    at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1+AsyncStateMachineBox`1[[System.Threading.Tasks.VoidTaskResult, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c+<<AsyncAction_FatalError_AV>b__6_1>d, Microsoft.DotNet.RemoteExecutor.Tests, Version=42.42.42.42, Culture=neutral, PublicKeyToken=null]].MoveNext()
>    at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Runtime.CompilerServices.IAsyncStateMachineBox, Boolean)
>    at System.Threading.Tasks.Task.RunContinuations(System.Object)
>    at System.Threading.Tasks.Task.TrySetResult()
>    at System.Threading.Tasks.Task+DelayPromise.CompleteTimedOut()
>    at System.Threading.TimerQueueTimer.Fire(Boolean)
>    at System.Threading.TimerQueue.FireNextTimers()
>    at System.Threading.UnmanagedThreadPoolWorkItem.System.Threading.IThreadPoolWorkItem.Execute()
>    at System.Threading.ThreadPoolWorkQueue.Dispatch()
>    at System.Threading.PortableThreadPool+WorkerThread.WorkerThreadStart()
> Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
> Expected: True
> Actual:   False
>    at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
>    at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
>    at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncFunc_ThrowException>b__3_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 54
> --- End of stack trace from previous location ---
>    at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 61
> --- End of stack trace from previous location ---
>    at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
> Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
> Expected: True
> Actual:   False
>    at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
>    at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
>    at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncAction_ThrowException>b__1_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 29
> --- End of stack trace from previous location ---
>    at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 65
> --- End of stack trace from previous location ---
>    at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
> Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
>    at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c.<FatalError_AV>b__8_1()
>    at System.RuntimeMethodHandle.InvokeMethod(System.Object, Void**, System.Signature, Boolean)
>    at System.Reflection.MethodInvoker.Invoke(System.Object, IntPtr*, System.Reflection.BindingFlags)
>    at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
>    at Microsoft.DotNet.RemoteExecutor.Program.Main(System.String[])
>   Finished:    Microsoft.DotNet.RemoteExecutor.Tests
> ```

@elinor-fung it looks like the exception is different. Here is a passing run:

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-arcade-refs-pull-9628-merge-7f71b0fd05b7464fae/Microsoft.DotNet.RemoteExecutor.Tests.dll/1/console.018d90f7.log?helixlogtype=result

```
xUnit.net Console Runner v2.4.2-pre.9+e627c231a5 (64-bit .NET 6.0.5)
  Discovering: Microsoft.DotNet.RemoteExecutor.Tests
  Discovered:  Microsoft.DotNet.RemoteExecutor.Tests
  Starting:    Microsoft.DotNet.RemoteExecutor.Tests
Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
   at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncFunc_ThrowException>b__2_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 40
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 61
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
   at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncAction_ThrowException>b__0_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 19
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 65
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
  Finished:    Microsoft.DotNet.RemoteExecutor.Tests
=== TEST EXECUTION SUMMARY ===
   Microsoft.DotNet.RemoteExecutor.Tests  Total: 5, Errors: 0, Failed: 0, Skipped: 0, Time: 0.505s
```

**Comment by elinor-fung** [comment]
There are multiple tests that intentionally make the launched process crash. Both crashes from an unhandled managed and an AV are intentional.

From logs from this PR in job that passed:
https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-arcade-refs-pull-9319-merge-1b4c5151dea146cfa7/Microsoft.DotNet.RemoteExecutor.Tests.dll/1/console.49af8d0f.log?helixlogtype=result

AV
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests+<>c.<FatalError_AV>b__8_1()
   at System.RuntimeMethodHandle.InvokeMethod(System.Object, Void**, System.Signature, Boolean)
   at System.Reflection.MethodInvoker.Invoke(System.Object, IntPtr*, System.Reflection.BindingFlags)
   at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
   at Microsoft.DotNet.RemoteExecutor.Program.Main(System.String[])
```
https://github.com/dotnet/arcade/blob/07a25f4424060df1c35e9947094405bc6f314f9d/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs#L119

Unhandled managed exception:
```
Unhandled exception. Xunit.Sdk.TrueException: Assert.True() Failure
Expected: True
Actual:   False
   at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 95
   at Xunit.Assert.True(Boolean condition) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 62
   at Microsoft.DotNet.RemoteExecutor.Tests.RemoteExecutorTests.<>c.<<AsyncFunc_ThrowException>b__3_1>d.MoveNext() in /_/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs:line 54
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 61
--- End of stack trace from previous location ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args) in /_/src/Microsoft.DotNet.RemoteExecutor/src/Program.cs:line 94
```
https://github.com/dotnet/arcade/blob/07a25f4424060df1c35e9947094405bc6f314f9d/src/Microsoft.DotNet.RemoteExecutor/tests/RemoteExecutorTests.cs#L54

Passing run on redhat from a different PR that shows the same expected errors: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-arcade-refs-pull-9680-merge-42d4f3224bb14207a7/Microsoft.DotNet.RemoteExecutor.Tests.dll/1/console.e14576ca.log?helixlogtype=result

The intent of those tests is to crash the launched process and make sure it gets reported / detected as a failure. I guess something with the update to Preview 5 in this PR is resulting in the work item (only redhat?) failing even though the logs indicate the tests all passed. Where would we look for more information on what happened after all the tests ran/passed to make the work item fail?

**Comment by MattGal** [comment]
@elinor-fung this fails in a fascinating way so I am taking a peek.

**Comment by adiaaida** [comment]
It started in at least preview 4 (the previous version of this PR to upgrade from preview 2 to preview 4 was failing for the same reason). I don't know if preview 3 was affected, since we didn't try a preview 3 build. @dsplaisted @marcpopMSFT, do either of you know if anything changed in the sdk that might cause this sort of issue (the tests say they are passing but then dotnet test actually fails)? This is only happening, as far as we can see, in these RemoteExecutor tests.

**Comment by MattGal** [comment]
@adiaaida / @elinor-fung it's running and passing, then literally filling the disk on the machine with dumps.  Is it possible dumps got larger in this time frame?  (trying to figure out what the available disk space was when it started)

```
=== TEST EXECUTION SUMMARY ===
   Microsoft.DotNet.RemoteExecutor.Tests  Total: 11, Errors: 0, Failed: 0, Skipped: 0, Time: 10.896s
+ export _commandExitCode=0
+ _commandExitCode=0
+ /bin/python3 /mnt/work/B14F0934/p/reporter/run.py https://dev.azure.com/dnceng/ public 48352892 eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Im9PdmN6NU1fN3AtSGpJS2xGWHo5M3VfVjBabyJ9.eyJuYW1laWQiOiJlNDI4NWM4Yy0zZmQ1LTQyNjctOWIxYy00MjE5NWM0N2E1NTMiLCJzY3AiOiJMb2NhdGlvblNlcnZpY2UuQ29ubmVjdCBQaXBlbGluZUNhY2hlLlJlYWRXcml0ZVJvb3RBY2Nlc3MgUmVhZEFuZFB1Ymxpc2hUZXN0OjllZTZkNDc4LWQyODgtNDdmNy1hYWNjLWY2ZTZkMDgyYWU2ZCBSZWFkQW5kVXBkYXRlQnVpbGRCeVVyaTo5ZWU2ZDQ3OC1kMjg4LTQ3ZjctYWFjYy1mNmU2ZDA4MmFlNmQvZG90bmV0L2FyY2FkZS8yMDg6QnVpbGQvQnVpbGQvMTgyNDAzMSIsImF1aSI6ImQxZTkzOWQ3LTJmY2UtNDdjZi05ODFmLWQ2MzRkOWRkMmI3YyIsInNpZCI6ImRhYTk0MmUxLTE2OWItNGQyNy1hYTk5LTcyNDBhMDZkN2I3MSIsIkJ1aWxkSWQiOiI5ZWU2ZDQ3OC1kMjg4LTQ3ZjctYWFjYy1mNmU2ZDA4MmFlNmQ7MTgyNDAzMSIsInBwaWQiOiJ2c3RmczovLy9CdWlsZC9CdWlsZC8xODI0MDMxIiwib3JjaGlkIjoiMDZhNDE0MjYtZjViMi00YzVhLTkwZmUtMWM1OWRhMzY1Zjc5LnRlc3Qud2luZG93c19udC5idWlsZF9yZWxlYXNlIiwiaXNzIjoiYXBwLnZzdG9rZW4udmlzdWFsc3R1ZGlvLmNvbSIsImF1ZCI6ImFwcC52c3Rva2VuLnZpc3VhbHN0dWRpby5jb218dnNvOmI1NWRlNGVkLTRiNWEtNDIxNS1hOGU0LTBhMGE1ZjcxZTdkOCIsIm5iZiI6MTY1NTIyODQ3NiwiZXhwIjoxNjU1MjM1MDc2fQ.nyGiQGQPnJUjtJsnSSf1vd5ciICj1eeAiAZMApOzeAOsuU6EX6pKVa8VuD6tRZMXEFuHaJ7qJHDRDhtmhcYXOWJ4sDI97CoLvB3eklvPQKJ2aluBvCamkRXaJFbKimfH8CV0YVhjD_ORRBa-8BfTjh01zVdxO7EVBvIm9AcUZ15q5YtNaIK9tA8Tvvn9LdBYqteAj6OE6PK8vQpwIuXZ6jfcguEgMLITygQWgnj4G_q21ztNuywkpzxFZ9MXz5VX-5ppODc4K0_jFejkw91eTmcM5ApnVrmOtGrcKy8b2aDZJSHAjjol3yQYBhWs9Y6WdLVlYK75vqmBBgGg-aLyPA
2022-06-14T18:10:06.484Z	INFO   	run.py	run(48)	main	Beginning reading of test results.
2022-06-14T18:10:06.485Z	INFO   	run.py	__init__(42)	read_results	Searching '/mnt/work/B14F0934/w/96C2084F/e' for test results files
2022-06-14T18:10:06.485Z	INFO   	run.py	__init__(48)	read_results	Found results file /mnt/work/B14F0934/w/96C2084F/e/testResults.xml with format xunit
2022-06-14T18:10:06.486Z	INFO   	run.py	__init__(42)	read_results	Searching '/mnt/work/B14F0934/w/96C2084F/uploads' for test results files
2022-06-14T18:10:06.486Z	INFO   	run.py	packing_test_reporter(30)	report_results	Packing 11 test reports to '/mnt/work/B14F0934/w/96C2084F/e/__test_report.json'
2022-06-14T18:10:06.492Z	INFO   	run.py	packing_test_reporter(33)	report_results	Packed 4569 bytes
--- Logging error ---
Traceback (most recent call last):
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/__init__.py", line 998, in emit
    self.flush()
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/__init__.py", line 978, in flush
    self.stream.flush()
OSError: [Errno 28] No space left on device
Call stack:
  File "/mnt/work/B14F0934/p/reporter/run.py", line 68, in <module>
    main()
  File "/mnt/work/B14F0934/p/reporter/run.py", line 65, in main
    reporter.report_results(all_results)
  File "/etc/helix/scripts/helix/test_reporting/packing_test_reporter.py", line 30, in report_results
    self._log.info("Packing %d test reports to '%s'", len(results), path)
Message: "Packing %d test reports to '%s'"
Arguments: (11, '/mnt/work/B14F0934/w/96C2084F/e/__test_report.json')
--- Logging error ---
Traceback (most recent call last):
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/handlers.py", line 71, in emit
    if self.shouldRollover(record):
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/handlers.py", line 188, in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
OSError: [Errno 28] No space left on device
Call stack:
  File "/mnt/work/B14F0934/p/reporter/run.py", line 68, in <module>
    main()
  File "/mnt/work/B14F0934/p/reporter/run.py", line 65, in main
    reporter.report_results(all_results)
  File "/etc/helix/scripts/helix/test_reporting/packing_test_reporter.py", line 33, in report_results
    self._log.info("Packed %d bytes", os.path.getsize(path))
Message: 'Packed %d bytes'
Arguments: (4569,)
+ exit 0
```

**Comment by adiaaida** [comment]
Wait.... this is suspicious:

```
--- Logging error ---
Traceback (most recent call last):
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/__init__.py", line 998, in emit
    self.flush()
  File "/opt/rh/rh-python36/root/usr/lib64/python3.6/logging/__init__.py", line 978, in flush
    self.stream.flush()
OSError: [Errno 28] No space left on device
Call stack:
  File "/mnt/work/B14F0934/p/reporter/run.py", line 68, in <module>
    main()
  File "/mnt/work/B14F0934/p/reporter/run.py", line 65, in main
    reporter.report_results(all_results)
  File "/etc/helix/scripts/helix/test_reporting/packing_test_reporter.py", line 30, in report_results
    self._log.info("Packing %d test reports to '%s'", len(results), path)
Message: "Packing %d test reports to '%s'"
Arguments: (11, '/mnt/work/B14F0934/w/96C2084F/e/__test_report.json')
```

In particular `OSError: [Errno 28] No space left on device`

It's not failing on main, so I think that means the helix machines haven't changed. But maybe something else got bigger between preview.2 and preview.4? File size, log size, dump size?

Also FYI @MattGal

**Comment by jkoritzinsky** [comment]
> it's running and passing, then literally filling the disk on the machine with dumps. Is it possible dumps got larger in this time frame? (trying to figure out what the available disk space was when it started)

Yes, IIRC dumps got larger due to the change to turn on GC Regions.

**Comment by mmitche** [comment]
@MattGal What do you think the right direction is here given that the dumps are larger?

**Comment by ViktorHofer** [comment]
Note that dotnet/runtime already independently upgraded to 7.0 Preview 5.

**Comment by MattGal** [comment]
> @MattGal What do you think the right direction is here given that the dumps are larger?

I think the dumps being larger was a red herring and that this was just an unlucky run since every out-of-space message came from the same machine... it may have literally even been one where the /mnt drive failed to exist.  

I think if it passes on re-run it's fine, and I wrote a (low pri) issue about getting the log directory onto the /mnt drive at a later date, but this is not urgent nor may never even be done given that it's only happened the once, so far.

**Comment by adiaaida** [comment]
I don't think this is going to pass on retry. It was consistently failing with preview 4 and has continued with preview 5.

**Comment by MattGal** [comment]
> I don't think this is going to pass on retry. It was consistently failing with preview 4 and has continued with preview 5.

I see.   I'll keep poking for a solution here, but there isn't likely going to be a quick one if we have to increase space for logging.

**Comment by hoyosjs** [comment]
If the remote executor has an expectation for things to fail, likely adding an extension that prevents dumps from being taken - or to use something like triage dumps - makes sense.

**Comment by MattGal** [comment]
@mmitche : this commit will unblock your PR, and remove a queue that's going to die in 10 days: https://github.com/MattGal/arcade/commit/f86e8f135d1f3006bc519ecf816bfc15c5957a06

**Comment by MattGal** [comment]
With @mmitche OOF I prepped my commit as a separate PR in case this change is still desired : https://github.com/dotnet/arcade/pull/9769

**Pull request closed without merging**